compile failed for the famfs as the extent_type name has been changed to famfs_extent_type.

compile failed as the error:
#make all
make[3]: Entering directory '/root/famfs/debug'
[  3%] Building C object CMakeFiles/libfamfs.dir/src/famfs_lib.c.o
/root/famfs/src/famfs_lib.c:423:28: warning: ‘enum extent_type’ declared inside parameter list will not be visible outside of this definition or declaration
  423 |                       enum extent_type *type)
      |                            ^~~~~~~~~~~
/root/famfs/src/famfs_lib.c:421:1: error: conflicting types for ‘famfs_get_device_size’; have ‘int(const char *, size_t *, enum extent_type *)’ {aka ‘int(const char *, long unsigned int *, enum extent_type *)’}
  421 | famfs_get_device_size(const char       *fname,
      | ^~~~~~~~~~~~~~~~~~~~~
In file included from /root/famfs/src/famfs_lib.c:35:
/root/famfs/src/famfs_lib.h:21:12: note: previous declaration of ‘famfs_get_device_size’ with type ‘int(const char *, size_t *, enum famfs_extent_type *)’ {aka ‘int(const char *, long unsigned int *, enum famfs_extent_type *)’}
   21 | extern int famfs_get_device_size(const char *fname, size_t *size, enum famfs_extent_type *type);
      |            ^~~~~~~~~~~~~~~~~~~~~
/root/famfs/src/famfs_lib.c: In function ‘famfs_mkfs’:
/root/famfs/src/famfs_lib.c:3812:14: error: variable ‘type’ has initializer but incomplete type
 3812 |         enum extent_type type = SIMPLE_DAX_EXTENT;
      |              ^~~~~~~~~~~
/root/famfs/src/famfs_lib.c:3812:26: error: storage size of ‘type’ isn’t known
 3812 |         enum extent_type type = SIMPLE_DAX_EXTENT;
      |                          ^~~~
/root/famfs/src/famfs_lib.c:3812:26: warning: unused variable ‘type’ [-Wunused-variable]
make[3]: *** [CMakeFiles/libfamfs.dir/build.make:76: CMakeFiles/libfamfs.dir/src/famfs_lib.c.o] Error 1
make[3]: Leaving directory '/root/famfs/debug'
make[2]: *** [CMakeFiles/Makefile2:184: CMakeFiles/libfamfs.dir/all] Error 2
make[2]: Leaving directory '/root/famfs/debug'
make[1]: *** [Makefile:146: all] Error 2
make[1]: Leaving directory '/root/famfs/debug'
make: *** [Makefile:13: debug] Error 2

--- a/linux_include/famfs_ioctl.h
+++ b/linux_include/famfs_ioctl.h
@@ -14,11 +14,12 @@
 #include <linux/ioctl.h>
 #include <linux/uuid.h>
 
+#define FAMFS_KABI_VERSION 42
 #define FAMFS_MAX_EXTENTS 2
 
 /* We anticipate the possiblity of supporting additional types of extents */
-enum extent_type {
-	SIMPLE_DAX_EXTENT = 13,
+enum famfs_extent_type {
+	SIMPLE_DAX_EXTENT,
 	INVALID_EXTENT_TYPE,
 };
 
@@ -42,7 +43,7 @@ enum famfs_file_type {
  * @ext_list: 1 or more extents
  */
 struct famfs_ioc_map {
-	enum extent_type          extent_type;
+	enum famfs_extent_type    extent_type;
 	enum famfs_file_type      file_type;
 	__u64                     file_size;
 	__u64                     ext_list_count;

--- a/src/famfs_lib.c
+++ b/src/famfs_lib.c
@@ -1085,7 +1085,7 @@ famfs_mkmeta(const char *devname)
 		return -1;
 	}
 
-	if (file_has_map(sbfd)) {
+	if (file_has_map(logfd)) {
 		fprintf(stderr, "%s: found valid log file; doing nothing\n", __func__);
 	} else {
 		ext.famfs_extent_offset = sb->ts_log_offset;

--- a/src/famfs_lib.c
+++ b/src/famfs_lib.c
@@ -420,7 +420,7 @@ famfs_get_role_by_path(const char *path, uuid_le *fs_uuid_out)
 int
 famfs_get_device_size(const char       *fname,
 		      size_t           *size,
-		      enum extent_type *type)
+		      enum famfs_extent_type *type)
 {
 	char spath[PATH_MAX];
 	char *basename;
@@ -3809,7 +3809,7 @@ famfs_mkfs(const char *daxdev,
 {
 	int rc;
 	size_t devsize;
-	enum extent_type type = SIMPLE_DAX_EXTENT;
+	enum famfs_extent_type type = SIMPLE_DAX_EXTENT;
 	struct famfs_superblock *sb;
 	struct famfs_log *logp;
 	u64 min_devsize = 4 * 1024ll * 1024ll * 1024ll;

--- a/src/famfs_lib.h
+++ b/src/famfs_lib.h
@@ -8,6 +8,7 @@
 
 #include <linux/uuid.h> /* Our preferred UUID format */
 #include <uuid/uuid.h>  /* for uuid_generate / libuuid */
+#include <linux/famfs_ioctl.h>
 
 #include "famfs.h"
 #include "famfs_meta.h"
@@ -15,10 +16,18 @@
 #define SB_FILE_RELPATH    ".meta/.superblock"
 #define LOG_FILE_RELPATH   ".meta/.log"
 
+/* Hack due to unintended consequences of kmod v1/v2 change */
+#ifndef FAMFS_KABI_VERSION
+enum famfs_extent_type {
+	SIMPLE_DAX_EXTENT,
+	INVALID_EXTENT_TYPE,
+};
+#endif
+
 int famfs_module_loaded(int verbose);
 void *famfs_mmap_whole_file(const char *fname, int read_only, size_t *sizep);
 
-extern int famfs_get_device_size(const char *fname, size_t *size, enum extent_type *type);
+extern int famfs_get_device_size(const char *fname, size_t *size, enum famfs_extent_type *type);
 int famfs_check_super(const struct famfs_superblock *sb);
 int famfs_fsck(const char *devname, int use_mmap, int human, int verbose);
 

--- a/test/famfs_unit.cpp
+++ b/test/famfs_unit.cpp
@@ -250,7 +250,7 @@ TEST(famfs, famfs_open_relpath)
 
 TEST(famfs, famfs_get_device_size)
 {
-	enum extent_type type;
+	enum famfs_extent_type type;
 	size_t size;
 	int rc;
 

